EIPS/eip-2135.md

```diff
@@ -1,80 +1,103 @@
+Being a digital assets sometimes means a consumable power. One most common seen examples would be a concert ticket.
+It will be "consumed" at the moment the ticket-holder uses the ticket to get access to enter a concert.
 
 By having a standard ERC interface, the Ethereum ecosystem can interoperate to provide services, clients, UI, and inter-contract functionalities on top of this very general use-case.
 
 ## Specification
-The standard will mainly contain the following interface.
 
-### The required interface
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+1. Any compliant contract **MUST** implement the following interface
 
 ```solidity
 pragma solidity >=0.7.0 <0.9.0;
 
-interface EIP2135 {
-
-  // The main consume function
-  function consume(uint256 _assetId) external returns(bool _success);
-
-  // The interface to check whether an asset is consumable.
-  function isConsumable(uint256 _assetId) external view returns (bool _consumable);
-
-  // The interface to check whether an asset is consumable.
-  event OnConsumption(uint256 indexed assetId);
+/// The EIP-165 identifier of this interface is 0x0c44c20d
+interface IEIP2135 {
+
+  /// @notice The consume function consumes a token every time it succeeds.
+  /// @param _consumer the address of consumer of this token. It doesn't have
+  ///                  to be the EOA or contract Account that initiates the TX.
+  /// @param _assetId  the NFT asset being consumed
+  /// @param _data     extra data passed in for consume for extra message
+  ///                  or future extension.
+  function consume(address _consumer, uint256 _assetId, bytes[] calldata _data) external returns(bool _success);
+
+  /// @notice The interface to check whether an asset is consumable.
+  /// @param _consumer the address of consumer of this token. It doesn't have
+  ///                  to be the EOA or contract Account that initiates the TX.
+  /// @param _assetId  the NFT asset being consumed
+  function isConsumableBy(address _consumer, uint256 _assetId) external view returns (bool _consumable);
+
+  /// @notice The event emitted when there is a successful consumption.
+  /// @param _consumer the address of consumer of this token. It doesn't have
+  ///                  to be the EOA or contract Account that initiates the TX.
+  /// @param _assetId  the NFT asset being consumed
+  /// @param _data     extra data passed in for consume for extra message
+  ///                  or future extension.
+  event OnConsumption(address indexed _consumer, uint256 indexed _assetId, bytes[] _data);
 }
 ```
 
+2. If the compliant contract is an [EIP-721](./eip-721.md) or [EIP-1155](./eip-1155.md), in addition to `OnConsumption`, it **MUST** also emit related `Transfer` / `TransferSingle` event as if a token has been transfer from the current holder to `0x00` address
```

Comment:
Is it required to implement one of EIP-721 or EIP-1155, or is that optional?

Reply:
Good question! I added an answer in rationale section